.github/workflows/build-and-test.yml

```diff
@@ -36,3 +36,59 @@ jobs:
       # If any test fails, the job and workflow will fail.
       - name: Test ${{ matrix.target_framework }}
         run: dotnet test --configuration Release --no-build --verbosity normal --framework ${{ matrix.target_framework }}
+      
+      # Using GitHub upload-artifact action to persist the workspace for the next job.
+      - name: Upload workspace
+        uses: actions/upload-artifact@v3
+        with:
+          name: workspace
+          path: ./
+
+  # code_analysis job runs code analysis scans against the codebase looking for vulnerabilities.
```

Comment:
We should look into reusing workflows.

Reply:
I think @JacobStade1 had tried that initially and had some trouble getting it to work.  Maybe he can comment more on it.  That being said, looking at the documentation about it, I think it would be a good idea to try again.

Reply:
I was able to get them to work, but the issue I encountered was that the reusable workflow could not be called from a job that had steps. Meaning that you could not do preparation work before calling the workflow, unless it was part of a different job.

It makes sense to use it here, as that limitation should not cause issues.